carbon-design-system/carbon
# fix(10343): Adds support for 'required' prop on RadioTile & TileGroup

Closes #10343

Adds support for '`required`' prop on `RadioTile`
Adds support for '`required`' prop on `TileGroup`


**New**
`RadioTile` supports '`required`' prop and attaches it to underlying `Input`
Adds new `TestCase` to verify this functionality 
`TileGroup` supports '`required`' prop and attaches it to children `RadioTile`



**Changed**
`RadioTile` accepts `required` prop 
`PublicAPI` snapshot updated with 'required' boolean attribute 
The `required` attribute has to be specified only on every `TileGroup` instead of on every child `RadioTile`


#### Testing / Reviewing

1. Add TileGroup with required RadioTiles to a form like shown below 
2. Submit form without selecting any of the tiles.

(Take pull of this PR and update any one of the storybook example with below code )

```
export const Test = () => {
  function onSubmit(e) {
    e.preventDefault();
    alert('submitted');
  }

  return (
    <div style={{ padding: '2em' }}>
      <Form onSubmit={onSubmit}>
        <TileGroup name="radio" required>
          <RadioTile value="1">
            Tile 1
          </RadioTile>
          <RadioTile value="2">
            Tile 2
          </RadioTile>
          <RadioTile value="3">
            Tile 3
          </RadioTile>
        </TileGroup>
        <Button type="submit">Submit</Button>
      </Form>
    </div>
  );
};
```
3. Verify the repro of below behaviour   
<img width="662" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/1ff434d6-b803-4086-836b-5ad670d55c83">
